src/components/CountdownTimerContainer/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect } from 'react'
+import './styles.css'
+import PropTypes from 'prop-types'
+import CountdownTimer from '../CountdownTimer'
+import dayjs from 'dayjs'
+import duration from 'dayjs/plugin/duration'
+import relativeTime from 'dayjs/plugin/relativeTime'
+import updateLocale from 'dayjs/plugin/updateLocale'
+
+const CountdownTimerContainer = ({ expirationDate }) => {
+  const [timeLeft, setTimeLeft] = useState('00:00:00')
+
+  const setTimeFormatted = (expirationDate) => {
+    const timeLeftInDuration = getTimeLeftInDuration(expirationDate)
+    let time = ''
+    if (timeLeftInDuration.asDays() > 1) {
+      time = getRelativeTimeToExpirationTime(expirationDate)
+    } else {
+      time = getDurationTimeInHHMMSSFormat(timeLeftInDuration)
+    }
+
+    return timeLeft !== 0 ? () => setTimeLeft(time) : finishTimer()
+  }
+
+  useEffect(() => {
+    const timer = setInterval(setTimeFormatted(expirationDate), 1000)
+    return () => clearInterval(timer)
+  })
```

Comment:
Here I would do a bit of refactoring:

```suggestion
  const [timeLeft, setTimeLeft] = useState(() => getFormattedTimeLeft(expirationDate))

  const getFormattedTimeLeft = (expirationDate) => {
    const timeLeftInDuration = getTimeLeftInDuration(expirationDate)

    if (timeLeftInDuration.asMilliseconds() <= 0) return '00:00:00'
    if (timeLeftInDuration.asDays() > 1) return getRelativeTimeToExpirationTime(expirationDate)

    return getDurationTimeInHHMMSSFormat(timeLeftInDuration)
  }

  useEffect(() => {
    const calculateAndSetTimeLeft = () => setTimeLeft(getFormattedTimeLeft(expirationDate))
    const intervalID = setInterval(calculateAndSetTimeLeft, 1000)

    return () => clearInterval(intervalID)
  }, [expirationDate])
```

Reply:
Also, `getFormattedTimeLeft` is a function that you can define outside the React component, thus getting declared once instead of on each render.